secure-ios-app/authentication/AuthenticationViewController.swift

```diff
@@ -11,17 +11,25 @@ import UIKit
 protocol AuthListener {
     func startAuth(presentingViewController: UIViewController)
     func logout()
+    func performPreCertCheck(onCompleted: @escaping (Bool) -> Void)
 }
 
 /* The view controller for the authentication view. It should pass the user events to the listener (interactor) */
 class AuthenticationViewController: UIViewController {
     
     @IBOutlet weak var authenticationButton: UIButton!
-    
+    @IBOutlet weak var logoImage: UIImageView!
+    @IBOutlet weak var backgroundImage: UIImageView!
+    @IBOutlet weak var certPinningError: UILabel!
+    @IBOutlet weak var dangerLogo: UIImageView!
+
     var authListener: AuthListener?
 
     override func viewDidLoad() {
         super.viewDidLoad()
+        
+        // perform cert pinning of the auth server on load
+        self.performPreCertCheck()
```

Comment:
this means the cert check will only happen on the view initialisation time. But since we are using a singleton object of the view controller, it means this will only be called on the very first time when the view is displayed. If user go to another view and come back, this check will not happen.

I managed to capture some https traffic using the proxy tool:
1. Disable proxy. Go to authentication page, login and then logout. But stay in the authentication page.2
2. Start the proxy tool and enable SSL proxy
3. I can authentication again and capture the data.

I think it's best to perform the check when the 'Authentication' button is clicked. This will make sure the cert pinning check will ALWAYS happen

Reply:
The fact that our auth screen isn't fronting the app creates a bit of a weird use case and isn't inline with typical real world apps, meaning this will be possible. Security sensitive apps like AIB and other banking apps will perform preflight checks on app load to check if the channel is secure before allowing any user interaction, and prevent access with pinning failures etc. Since our auth screen isn't fronting the app, the same technique doesn't work that well for our app is designed. I have made the change anyway. 

I can add a quick note in the docs for how it might be better to handle for real world apps where the auth screen is fronting the application.